### PR TITLE
[Core] Error info pubsub (Remove ray.errors API)

### PR DESCRIPTION
**Pull request by kisuke95**
## Why are these changes needed?

Access job error info through pub/sub

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Comment by rkooo567** [comment]
It is great we can remove lots of code! But it looks like some of tests still fail (especially `test_failure.py`). Can you ping me again once they are resolved?

File: src/ray/gcs/gcs_server/gcs_node_manager.cc
Comment (kisuke95) [inline src/ray/gcs/gcs_server/gcs_node_manager.cc:369]: @rkooo567  `error_info_accessor` in a member of `redis_accessor`, it causes the test_failure.py failed

**Comment by rkooo567** [comment]
@kisuke95 Can we merge this PR after our branch cut (this Wed). I think it could be breaking changes, so I'd like to avoid merging it right before branch cut.

File: python/ray/tests/test_actor_failures.py
Comment (rkooo567) [inline python/ray/tests/test_actor_failures.py:24]: Why don't we move these to test utils instead?

File: python/ray/tests/test_actor_failures.py
Comment (rkooo567) [inline python/ray/tests/test_actor_failures.py:41]: Remove these lines!

File: python/ray/tests/test_actor_failures.py
Comment (rkooo567) [inline python/ray/tests/test_actor_failures.py:656]: Can we move it to fixture instead of repeating?

Comment (rkooo567) [reply python/ray/tests/test_actor_failures.py:656]: Maybe we can also add unsubscribe in the fixture.

File: python/ray/tests/test_actor_failures.py
Comment (rkooo567) [inline python/ray/tests/test_actor_failures.py:736]: Why do we wait for 2 msgs?

File: python/ray/tests/test_advanced_3.py
Comment (rkooo567) [inline python/ray/tests/test_advanced_3.py:360]: Why don't we just remove it?

File: python/ray/tests/test_failure.py
Comment (rkooo567) [inline python/ray/tests/test_failure.py:530]: What does `"This test does not work yet."` mean? We should fix this before merging this PR right?

File: python/ray/tests/test_failure.py
Comment (rkooo567) [inline python/ray/tests/test_failure.py:832]: Why is this removed?

File: python/ray/tests/test_failure.py
Comment (rkooo567) [inline python/ray/tests/test_failure.py:787]: Why is this removed?

File: python/ray/tests/test_multi_node.py
Comment (rkooo567) [inline python/ray/tests/test_multi_node.py:14]: We shouldn't remove this test. We should test if error msgs are still streamed to different drivers correctly.

File: python/ray/tests/test_stress_failure.py
Comment (rkooo567) [inline python/ray/tests/test_stress_failure.py:290]: ?

File: python/ray/tests/test_stress_failure.py
Comment (rkooo567) [inline python/ray/tests/test_stress_failure.py:210]: Why is it commented out?

**Review by rkooo567 — CHANGES_REQUESTED** [review]
This still doesn't seem to be completed state. There are many tests that are skipped without enough justification. I will switch it to WIP for now.

File: python/ray/tests/test_actor_failures.py
Comment (kisuke95) [reply python/ray/tests/test_actor_failures.py:656]: @rkooo567  I think we still need to repeat it even though we move it to fixture, because `p` is used in `get_error_message` fuction. I don't know much about Python, is there other way that we can avoid repeating it ?

File: python/ray/tests/test_actor_failures.py
Comment (kisuke95) [reply python/ray/tests/test_actor_failures.py:736]: 1 or 2 is same, it just timeout because there is no error message. I will change it to 1

File: python/ray/tests/test_actor_failures.py
Comment (kisuke95) [reply python/ray/tests/test_actor_failures.py:24]: fixed

File: python/ray/tests/test_actor_failures.py
Comment (kisuke95) [reply python/ray/tests/test_actor_failures.py:656]: Move it to fixture doesn't work, the subscibe must happed after `ray.init`, because the `init_pubsub` uses `ray.worker.global_worker.redis_client` (it is `None` if we move `init_pubsub` to fixture)

**Comment by kisuke95** [comment]
> @kisuke95 Can we merge this PR after our branch cut (this Wed). I think it could be breaking changes, so I'd like to avoid merging it right before branch cut.

Yes, it's ok.

File: python/ray/tests/test_failure.py
Comment (kisuke95) [reply python/ray/tests/test_failure.py:832]: Sorry for this, I will add it back

File: python/ray/tests/test_actor_failures.py
Comment (rkooo567) [inline python/ray/tests/test_actor_failures.py:24]: Can you change the function name as well? `init_error_pubsub()` might sound better.

File: python/ray/tests/test_failure.py
Comment (kisuke95) [reply python/ray/tests/test_failure.py:530]: `ray_start_object_store_memory` is the setup function and it raises a `ValueError`, this makes the test failed.

File: python/ray/tests/test_failure.py
Comment (rkooo567) [inline python/ray/tests/test_failure.py:37]: You can put 

```
@pytest.fixture
def error_pubsub():
    p = init_error_pubsub
    yield p
    p.close()

def test_something(ray_start, error_pubsub):
    # and here, you can access p through error_pubsub
```
I think this will gurantee pubsub is always called after ray.init

File: src/ray/gcs/gcs_client/service_based_accessor.cc
Comment (rkooo567) [inline src/ray/gcs/gcs_client/service_based_accessor.cc:1331]: Remove comments.

**Review by rkooo567 — APPROVED** [review]
LGTM! I think it is good to go as long as 

1. Test passes
2. You addressed my leftover comments (just 2 additional).

Also, please don't merge this PR until I tell you to! (until we branch cut).

**Review by raulchen — APPROVED** [review]
Looks pretty nice.

**Comment by kisuke95** [comment]
@rkooo567 Have you finished branch cut ? Can we merge this PR ?

**Comment by rkooo567** [comment]
@kisuke95 Feel free to merge it. I already finished branch cut.

**Pull request merged**
